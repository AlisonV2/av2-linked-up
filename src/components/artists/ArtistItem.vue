<template>
  <div class="card artist-card">
    <div class="card-overlay"></div>
    <div class="card-details fade-in-top text-center">
      <h5 class="card-title">{{ artist.name }}</h5>
      <p class="card-text">{{ artist.shop }}</p>
      <p class="card-text">{{ artist.city }} ({{ artist.zip }})</p>
    </div>
    <img
      :src="artist.thumbnail"
      class="card-img-top"
      :alt="artist.name"
    />
    <div class="card-body">
      <h5 class="card-title">{{ artist.name }}</h5>
    </div>
  </div>
</template>

<script>
/**
 * @type Component
 * @name ArtistItem
 * @vue-prop {object} artists
 */
export default {
  name: 'ArtistItem',
  props: ['artist'],
};
</script>

<style lang="scss" scoped>
.artist-card {
  .card-title {
    font-family: $accent-font;
    color: $primary;
    font-size: 1.2rem;
  }
  .card-text {
    color: $light;
  }
  .card-details {
    position: absolute;
    text-align: center;
    padding-left: 1em;
    padding-right: 1em;
    width: 100%;
    top: 50%;
    left: 50%;
    opacity: 0;
    transform: translate(-50%, -50%);
    transition: all 0.3s ease-in-out 0s;
    .card-title {
      margin-bottom: 2rem;
    }
  }
  .card-overlay {
    background: rgba(0, 0, 0, 0.8);
    position: absolute;
    height: 100%;
    width: 100%;
    left: 0;
    top: 0;
    bottom: 0;
    right: 0;
    opacity: 0;
    transition: all 0.4s ease-in-out 0s;
  }
}

.artist-card:hover {
  .card-details {
    display: block;
  }
  .card-overlay {
    opacity: 1;
  }
  .card-details {
    top: 50%;
    left: 50%;
    opacity: 1;
  }
}

.fade-in-top {top: 20%};
</style>
