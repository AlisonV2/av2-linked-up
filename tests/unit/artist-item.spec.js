import { shallowMount, RouterLinkStub } from '@vue/test-utils';
import ArtistItem from '@/components/artists/ArtistItem';

const artist = {
  id: 'mR9w90C9fMQfVHJWjGo1TrBxPuk1',
  name: 'Dirty Jonz',
  shop: 'Le Dernier Rituel',
  city: 'BÃ¨gles',
  zip: '33130',
};

const config = {
  propsData: {
    artist: artist,
  },
  globals: {
    components: {
      'router-link': RouterLinkStub,
    },
  },
};
/**
 * @module ArtistItemTest
 */
describe('ArtistItem.vue', () => {
  it('ArtistItem Snapshot', () => {
    const component = shallowMount(ArtistItem, config);
    expect(component.element).toMatchSnapshot();
  });

  it('Should render true', () => {
    const component = shallowMount(ArtistItem, config);
    const artistName = component.find(
      '[data-test="mR9w90C9fMQfVHJWjGo1TrBxPuk1"]'
    );
    expect(artistName.text()).toBe(artist.name);
  });
});
