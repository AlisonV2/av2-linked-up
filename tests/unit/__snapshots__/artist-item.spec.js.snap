// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistItem.vue ArtistItem Snapshot 1`] = `
<div
  class="card artist-card"
>
  <div
    class="card-overlay"
  />
  <div
    class="card-details fade-in-top text-center"
  >
    <h5
      class="card-title"
      data-test="mR9w90C9fMQfVHJWjGo1TrBxPuk1"
    >
      Dirty Jonz
    </h5>
    <p
      class="card-text"
    >
      Le Dernier Rituel
    </p>
    <p
      class="card-text"
    >
      BÃ¨gles (33130)
    </p>
  </div>
  <img
    alt="Dirty Jonz"
    class="card-img-top"
    src="https://via.placeholder.com/150"
  />
  <div
    class="card-body"
  >
    <h5
      class="card-title"
    >
      Dirty Jonz
    </h5>
  </div>
</div>
`;
